app/javascript/src/BackendApis/components/IndexPage.js

```diff
@@ -20,10 +45,158 @@ type Props = {
 }
 
 const BackendsIndexPage = (props: Props) => {
-  console.log(props)
+  console.log('THIS IS THE PROPS' + JSON.stringify(props))
+
+  const [passedInProps, setPassedInProps] = useState(props)
+  console.log('what is the state' + setPassedInProps)
+
+  const tableColumns = [
+    'Name',
+    'System name',
+    'Last updated',
+    'Private base URL',
+    'Linked products'
+  ]
+
+  const tableRows = props.backends.map((tableRow) => {
+    return {
+      cells: [
+        { title: <a href="/">{tableRow.name}</a> },
+        tableRow.system_name,
+        <span className="api-table-timestamp">{tableRow.updated_at}</span>,
+        tableRow.private_endpoint,
+        tableRow.products_count
+      ]
+    }
+  })
+
+  const linkToPage = (event, rowId, rowData, extra, actionNumber) => {
+    const path = passedInProps && passedInProps.backends[rowId].links[actionNumber].path
+    window.location.href = path
+  }
+
+  const tableActions = [
+    {
+      title: 'Edit',
+      onClick: (event, rowId, rowData, extra) => linkToPage(event, rowId, rowData, extra, 0)
+    },
+    {
+      title: 'Overview',
+      onClick: (event, rowId, rowData, extra) => linkToPage(event, rowId, rowData, extra, 1)
+    },
+    {
+      title: 'Analytics',
+      onClick: (event, rowId, rowData, extra) => linkToPage(event, rowId, rowData, extra, 2)
+    },
+    {
+      title: 'Methods and Metrics',
+      onClick: (event, rowId, rowData, extra) => linkToPage(event, rowId, rowData, extra, 3)
+    },
+    {
+      title: 'Mapping Rules',
+      onClick: (event, rowId, rowData, extra) => linkToPage(event, rowId, rowData, extra, 4)
+    }
+  ]
+
+  const url = new URL(window.location.href)
+  var perPage = url.searchParams.get('per_page')
+  var page = url.searchParams.get('page')
+
+  const selectPerPage = (_event, selectedPerPage) => {
+    url.searchParams.set('per_page', selectedPerPage)
+    url.searchParams.delete('page')
+    window.location.href = url.toString()
+  }
+
+  const goToNextPage = (_event, number) => {
+    url.searchParams.set('page', number)
+    window.location.href = url.toString()
+  }
+
+  const goToPreviousPage = (_event, number) => {
+    url.searchParams.set('page', number)
+    window.location.href = url.toString()
+  }
+
+  const onFirstClick = (_event, number) => {
+    url.searchParams.set('page', number)
+    window.location.href = url.toString()
+  }
+
+  const onLastClick = (_event, number) => {
+    url.searchParams.set('page', number)
+    window.location.href = url.toString()
+  }
```

Comment:
We can avoid com repetition here:
```suggestion
  const goToPage(page) => {
    url.searchParams.set('page', page)
    window.location.href = url.toString()
  }
```